.gitignore

```diff
@@ -1,27 +1,112 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# I don't konw.
 _trial_temp
-build
-build-pypy
-MANIFEST
-dist
 apidocs
 *,cover
 .testrepository
-*.pyc
-*.pyd
-*.pyo
-*.so
 *~
-*.swp
-*.swh
-*.swn
-*.swo
+*.sw[phno]
 docs/tutorial/index.html
 dulwich.egg-info/
+docs/api/*.txt
+dulwich.dist-info
+
+# Unit test / coverage reports
+htmlcov/
 .tox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
 .idea/
 .coverage
-htmlcov/
-docs/api/*.txt
+
+# Distribution / packaging
+.Python
+build
+build/
+develop-eggs/
+dist
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+*.manifest
+*.spec
+
+# Installer logs
+pip-log.txt
+pip-delete-this-directory.txt
+
+# Translations
+*.mo
+*.pot
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# pyenv
+.python-version
+
+# celery beat schedule file
+celerybeat-schedule
+
+# SageMath parsed files
+*.sage.py
+
+# Environments
+.env
+.venv
+env/
+venv/
+ENV/
+env.bak/
+venv.bak/
+
+# Spyder project settings
+.spyderproject
+.spyproject
+
+# Rope project settings
+.ropeproject
+
+# mkdocs documentation
+/site
+
+# mypy
 .mypy_cache/
-.eggs
-dulwich.dist-info
+build-pypy
```

Comment:
I made this carefully. Although it was rejected last time because of repetitions and unnecessary needs, I can add more Python common ignore lists if possible.

Reply:
Please don't add ignore lines for anything that isn't generated by dulwich itself. Ignore files for IDEs, etc that some people use belong in their per-user ignore file - not in the project-specific one.

Reply:
I think you will need to consider the development environment used by other developers. The common ignore list will not affect the loss of other developers or existing files, because these files must be ignored during development.

For example, I have used Environments to avoid conflicts between packages on different projects, but you did not ignore related directories or files, which made me a little troublesome in editing, because Git will always show that the directory has not been indexed.

Jupyter is an editor, also used for Python programming language development, and the number of people using it is quite large. In addition, I also help to translate your documents, you can see this URL, although I haven't really finished it yet.

https://timmy61109.blogspot.com/2020/11/gitpython.html

And before I found out that some files actually need to be ignored, but the result was not there, resulting in unnecessary changes to some files in the project.

I mean:

> These ignore lists or other general ignore lists will not cause adverse effects on the project, but will help other developers to facilitate their development.

But if you insist on keeping only the ignored items that need to be used, no problem, I will delete them.

Translation from Google Translate

Reply:
People who use those editors can edit their global gitignore file (~/.gitignores_global) and add the files generated by their IDEs there. Otherwise, *all* Python projects on GitHub would have to carry and keep up to date this overly long list of ignores for every single editor.

Reply:
> People who use those editors can edit their global gitignore file (~/.gitignores_global) and add the files generated by their IDEs there. Otherwise, _all_ Python projects on GitHub would have to carry and keep up to date this overly long list of ignores for every single editor.

Yes, that's why GitHub has a recommended ignore list. Provide all projects to quickly create a ignore list. What I provide here is provided by GitHub, not by me, and I made this request because I encountered the above-mentioned problem.

But I want to make it quickly, so I will follow your request, because what you said is not unreasonable, but some people rely on the ignore list in the project or related assistance in the development of the project.

Reply:
I will not delete the Jupyter-related ignore list here, because I think this is very useful.

In addition, this ignore list should actually be ignored, so according to the relevance of the programming language, a long ignore list is normal.